src/Expr.js

```diff
@@ -74,60 +74,109 @@ var specialCases = {
   uppercase: 'UpperCase',
 }
 
-var exprToString = function(expr, caller) {
-  var isExpr = function(e) {
-    return e instanceof Expr || util.checkInstanceHasProperty(e, '_isFaunaExpr')
-  }
+/**
+ *
+ * @param {Expr} expression A FQL expression
+ * @returns {Boolean} Returns true for valid expressions
+ */
+function isExpr(expression) {
+  return (
+    expression instanceof Expr ||
+    util.checkInstanceHasProperty(expression, '_isFaunaExpr')
+  )
+}
 
-  if (isExpr(expr)) {
-    if ('value' in expr) return expr.toString()
-    expr = expr.raw
-  }
+/**
+ *
+ * @param {Object} obj An object to print
+ * @returns {String} String representation of object
+ */
+function printObject(obj) {
+  return (
+    '{' +
+    Object.keys(obj)
+      .map(function(k) {
+        return k + ': ' + exprToString(obj[k])
+      })
+      .join(', ') +
+    '}'
+  )
+}
 
-  var type = typeof expr
+/**
+ *
+ * @param {Array} arr An array to print
+ * @param {Function} toStr Function used for stringification
+ * @returns {String} String representation of array
+ */
+function printArray(arr, toStr) {
+  return arr
+    .map(function(item) {
+      return toStr(item)
+    })
+    .join(', ')
+}
 
-  if (type === 'string') {
-    return JSON.stringify(expr)
-  }
+/**
+ *
+ * @param {String} fn A snake-case FQL function name
+ * @returns {String} The correpsonding camel-cased FQL function name
+ */
+function convertToCamelCase(fn) {
+  // For FQL functions with special formatting concerns, we
+  // use the specialCases object above to define their casing.
+  if (fn in specialCases) fn = specialCases[fn]
 
-  if (type === 'symbol' || type === 'number' || type === 'boolean') {
-    return expr.toString()
-  }
+  return fn
+    .split('_')
+    .map(function(str) {
+      return str.charAt(0).toUpperCase() + str.slice(1)
+    })
+    .join('')
+}
 
-  if (type === 'undefined') {
-    return 'undefined'
+/**
+ *
+ * @param {*} expr A FQL expression
+ * @returns {Boolean} Indicates if arguments should be reversed
+ */
+function shouldReverseArgs(expr) {
```

Comment:
also, once `exprToString` is finished, that function won't necessary, `filter` and `foreach` is already handled there, `map` should be too

Reply:
To make sure I understand, was this originally here to force the order of arguments for `filter`, `foreach` and `map` but now we don't need this function since our `if` statements are now handling constructing the queries properly?

Reply:
That’s correct

Reply:
Some tests are now failing if I remove this function call.

Reply:
you should add `map` to the if wagon

Reply:
Done 👍